lib/logwriter.c

```diff
@@ -1716,6 +1714,8 @@ log_writer_new(guint32 flags, GlobalConfig *cfg)
   g_static_mutex_init(&self->pending_proto_lock);
   self->pending_proto_cond = g_cond_new();
 
+  log_pipe_add_info((LogPipe *)self, "writer");
+
```

Comment:
using &self->super as the argument keeps it typesafe, it's usually better than the cast.

Reply:
Thanks for this tips. I have changed to `&self->super`.